django/db/models/fields/__init__.py

```diff
@@ -656,6 +656,15 @@ def rel_db_type(self, connection):
         """
         return self.db_type(connection)
 
+    def cast_db_type(self, connection):
+        """
+        Return the data type that will be used in the Cast() function.
+        """
+        if self.get_internal_type() in connection.ops.cast_data_types:
+            data = self.db_type_parameters(connection)
+            return connection.ops.cast_data_types.get(self.get_internal_type()) % data
```

Comment:
Maybe store the data type instead of getting it twice?

```python
db_type = connection.ops.cast_db_types(self.get_internal_type())
if db_type:
    return db_type % self.db_type_parameters(connection)
return self.db_type(connection)
```